openml/datasets/functions.py

```diff
@@ -499,6 +504,9 @@ def create_dataset(name, description, creator, contributor,
     else:
         attributes_ = attributes
 
+    if row_id_attribute is None and hasattr(data, "index"):
```

Comment:
what happens with the dataframe index if `row_id_attribute` is set to a value other than the index? Will it be ignored at upload time? Can this be added to comments and somehow tested?

Reply:
Could you quickly show a code snippet (even pseudo-code) to illustrate what you mean? I am not sure to see the use-case?

Reply:
I think @janvanrijn wanted to only point out a potential issue:
```
In [17]: dataset = openml.datasets.functions.create_dataset(
    ...:             name=name,
    ...:             description=description,
    ...:             creator=creator,
    ...:             contributor=None,
    ...:             collection_date=collection_date,
    ...:             language=language,
    ...:             licence=licence,
    ...:             default_target_attribute=default_target_attribute,
    ...:             ignore_attribute=None,
    ...:             citation=citation,
    ...:             attributes='auto',
    ...:             data=df,
    ...:             row_id_attribute='rnd_str',
    ...:             format=None,
    ...:             version_label='test',
    ...:             original_data_url=original_data_url,
    ...:             paper_url=paper_url,
    ...: )

In [18]: dataset.row_id_attribute
Out[18]: 'rnd_str'

In [19]: df.index.name
Out[19]: 'index'
```

Reply:
OK so the PR is wrong. The behaviour needs to be better defined:

* If data is an array and `row_id_attribute` is `None` then we let `row_id_attribute` at `None`.
* If data is an array and `row_id_attribute` is not `None`, we need to have a check that this is actually an attribute.
* If data is a dataframe and `row_id_attribute` is `None`, we need to infer it. If `df.index.name` is `None` then we are back to the first bullet. Otherwise, we take the name as `row_id_attribute` and we need to `df.reset_index()` to include the data into the array when converting to numpy `df.values`.